tests/components/labOrderEntry/LabEntryForm.test.jsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import LabEntryForm from '../../../app/js/components/labOrderEntry/LabEntryForm';
+
+let props;
+let mountedComponent;
+
+const getComponent = () => {
+  if (!mountedComponent) {
+    mountedComponent = shallow(<LabEntryForm />);
+  }
+  return mountedComponent;
+};
+
+const mockTest = [1, 2, 3];
+const mockPanel = { id: 1 };
+
+describe('Component: LabEntryForm', () => {
+  beforeEach(() => {
+    mountedComponent = undefined;
+  });
+
+  it('should render on initial setup', () => {
+    const component = getComponent();
+    expect(component).toMatchSnapshot();
+  });
+
+  it('should be able to select a category', () => {
+    const component = getComponent();
+    const categoryButton = component.find('#category-button').at(1); // click on the second category iwth id of 2
+    categoryButton.simulate('click', {
+      target: {}
+    });
+    expect(component.state().categoryId).toEqual(2);
+  });
+
+  it('should set the seleted tests state of the component', () => {
+    const component = getComponent();
+    const instance = component.instance();
+    instance.selectTests(mockTest);
+    expect(component.state().selectedTests.includes(1)).toBeTruthy();
+  });
+
+  it('should add a new test option if there is none', () => {
+    const component = getComponent();
+    const instance = component.instance();
+    instance.setState({
+      selectedTests: mockTest
+    });
+    let newTestId = 4;
+    instance.selectTest(newTestId);
+    expect(component.state().selectedTests.includes(4)).toBeTruthy();
+    instance.selectTest(newTestId);
+    expect(component.state().selectedTests.includes(4)).toBeFalsy();
+  });
```

Comment:
I think you are testing for more than `adding new test option` in this test block.
It would be better if different test blocks handle the different tests, remember that some people are going to read your tests to understand how your app is supposed to behave.

Reply:
It has to be in the same test block because it is been toggled to `click unclick` should add and remove. putting it in different test block will just reset the state and we don't want that.

Reply:
If that is the case, you can just update the description of the test block to reflect the assertions you are doing in the test block.

Reply:
Done description has been changed